java/Mekhmet-Efe_Adzhar/src/com/zoolatech/lecture7/tasks/_2/ThreadOperations.java

```diff
@@ -0,0 +1,112 @@
+package com.zoolatech.lecture7.tasks._2;
+
+class ThreadOperations {
+
+    public static void explicitLocks() throws InterruptedException {
```

Comment:
you can remove the code duplication for three methods if you define a common interface, allow all three classes to implement it and pass to the method an instance of the interface